### ISSUE TITLE
Non-existant remote branch breaks merge conflict check

### ISSUE THREAD

**Issue by anubisthejackle** [anubisthejackle/git-hooks#1]
ERROR:

```
fatal: ambiguous argument 'remote/branchthatdoesntexistonremote': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```
This error is shown if you attempt to push to a remote branch that does not currently exist. This happens due to the merge conflict check attempting to perform a diff against the remote branch, when that branch doesn't exist.

## Solution
The real solution here is to not check for merge conflicts on push, but on commit. If we move this to a pre-commit check, we can do our `git diff --cached` against HEAD instead of a remote branch. This is better, too, because it will stop us from committing a merge conflict to begin with.

**Comment by anubisthejackle** [comment]
Actually I just realized this may not have to do with the slashes, but may in fact have to do with the fact the remote branch did not exist before this push. This bears testing.

**Comment by anubisthejackle** [comment]
I have confirmed this is actually related to the remote branch not existing at the time of push. Updating issue.

**Comment by anubisthejackle** [comment]
Solved in #2
